internal/cli/filtering.go

```diff
@@ -148,7 +150,8 @@ func (f Filtering) NewNoteFindOpts(notebook *core.Notebook) (core.NoteFindOpts,
 		return opts, fmt.Errorf("the --exact-match (-e) option is deprecated, use --match-strategy=exact (-Me) instead")
 	}
 
-	opts.Match = opt.NewNotEmptyString(f.Match)
+	opts.Match = make([]string, len(f.Match))
+	copy(opts.Match, f.Match)
```

Comment:
Again, safety copy? Needed?

Reply:
I think this one should be a copy as `opts.Match = f.Match` would assign the `[]string{}` pointers - because slices are structs - and so changes could leak upwards?

Reply:
Right, any change in `opts.Match` would impact `f.Match`.